### PR TITLE
Chore: Added text component e2e tests (Issue/72)

### PR DESCRIPTION
**Pull request by lemmyadams**
addresses https://github.com/adaptlearning/adapt-contrib-text/issues/72

### CODE REVIEW

**Review by ethan-lp — APPROVED** [review]

**Review by eleanor-heath — APPROVED** [review]

File: test/e2e/text.cy.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+describe('Text', function () {
+  beforeEach(function () {
+    cy.getData().then(function (data) {
+      this.data = data;
+      cy.visit('/');
+    });
```
Comment (oliverfoster) [inline test/e2e/text.cy.js:6]: This does the same thing:
```suggestion
    cy.getData();
    cy.visit('/');
```
This isn't the await/async promise syntax, it's a bespoke cypress promise-like syntax for generating some execution order. `cy.getData()`, without the `then` subclause completes before `cy.visit('/')` I believe.

`cy.getData()` does `this.data = data` already at https://github.com/adaptlearning/adapt_framework/blob/ad3913fb8a173b5a2d06c1dd7df0f91a6ab0881e/test/e2e/commands.js#L29

File: test/e2e/text.cy.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  });
+
+  it('should display the text component', function () {
+    const textComponent = this.data.components.find((component) => component._id === 'c-15')
```
Comment (oliverfoster) [inline test/e2e/text.cy.js:10]: We are not testing against the vanilla course only.
Please find the first text component and navigate to it in a structure and id agnostic way.

**Review by oliverfoster — CHANGES_REQUESTED** [review]

File: test/e2e/text.cy.js (hunk 2 above)
Comment (lemmyadams) [reply test/e2e/text.cy.js]: I didn't know, I was in the process of updating it

File: test/e2e/text.cy.js (hunk 1 above)
Comment (oliverfoster) [reply test/e2e/text.cy.js]: You can navigate to a component in adapt by routing to `#/id/c-15` etc

**Review by eleanor-heath — APPROVED** [review]

File: test/e2e/text.cy.js (hunk 1 above)
Comment (oliverfoster) [reply test/e2e/text.cy.js]: `this.data` in each test has the shortcuts to subsets of the main array `this.data = [{ _id: "", ... }, ...]`:
`this.data.contentObjects, this.data.articles, this.data.blocks, this.data.components, ` you don't need to recreate these with filters. https://github.com/adaptlearning/adapt_framework/blob/ad3913fb8a173b5a2d06c1dd7df0f91a6ab0881e/test/e2e/commands.js#L32-L63

File: test/e2e/text.cy.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+describe('Text', function () {
+  beforeEach(function () {
+    cy.getData().then(function (data) {
+      this.pages = data.contentObjects.filter(item => item._type === 'page' && item._classes !== 'assessment');
+      this.articles = data.filter(item => item._type === 'article');
+      this.blocks = data.filter(item => item._type === 'block');
+      this.components = data.filter(item => item._type === 'component');
```
Comment (cahirodoherty-learningpool) [inline test/e2e/text.cy.js:7]:
```suggestion
```
These are already available via helpful getters under `this.data`, so these lines can be deleted and the references below just need updated, e.g. `this.components` should be `this.data.components`

**Comment by oliverfoster** [comment]
Really we need a way of routing to and rendering a single component in Adapt to make component tests viably reliable.

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Pull request merged**